pkg/init/cmd/rc.init/main.go

```diff
@@ -55,13 +55,15 @@ func mountSilent(source string, target string, fstype string, flags uintptr, dat
 func mkchar(path string, mode, major, minor uint32) {
 	// unix.Mknod only supports int dev numbers; this is ok for us
 	dev := int(unix.Mkdev(major, minor))
-	err := unix.Mknod(path, mode, dev)
+	err := unix.Mknod(path, mode|unix.S_IFCHR, dev)
 	if err != nil {
 		if err.Error() == "file exists" {
 			return
```

Comment:
Do we want to chmod in all cases (also in this case?) or only if we actually created?

Reply:
I would rather set the umask at the beginning of the code so the permissions are unaffected and then set it to 022 at the end for the children to inherit, rather than also calling chmod all over the place.

Reply:
Yes, I was doubting about that; I know we had/have some code in docker that changes umask, and it caused issues for processes that were running concurrently.

But I guess that's not a problem for this case (no other code should be running concurrently 😅)